src/fstext/kaldi-fst-io.h

```diff
@@ -131,6 +161,86 @@ class VectorFstTplHolder {
 // Now make the original VectorFstHolder as the typedef o VectorFstHolder<StdArc>.
 typedef VectorFstTplHolder<StdArc> VectorFstHolder;
 
+// A generic Fst Holder which is similar with VectorFstHolder. Depending on the
+// FstType(), it will take different actions. This version currently supports 
+// ConstFst<StdArc> or VectorFst<StdArc>
+// This is a Holder class with T = Fst<Arc>, that meets the requirements
+// of a Holder class as described in ../util/kaldi-holder.h. This enables us to
+// read/write collections of FSTs indexed by strings, using the Table concept (
+// see ../util/kaldi-table.h).
+template<class Arc>
+class FstTplHolder {
+ public:
+  typedef Fst<Arc> T;
+
+  FstTplHolder(): t_(NULL) { }
+
+  static bool Write(std::ostream &os, bool binary, const T &t);
+
+  // Depending on the fst type, we cope t into this holder.
+  void Copy(const T &t) {
+    Clear();
+    if(t.Type() == "const") {
+      t_ = new ConstFst<Arc>(t);
+    } else if( t.Type() == "vector") {
+      t_ = new VectorFst<Arc>(t);
+    } else {
+      KALDI_WARN << "The version currently only support VectorFst and ConstFst.";
```

Comment:
this should be an error not a warning (anyway the code t_ = &t would later cause a double free error).